src/wp-includes/class-wp-image-editor.php

```diff
@@ -336,11 +337,17 @@ protected function get_output_format( $filename = null, $mime_type = null ) {
 			// If no file specified, grab editor's current extension and mime-type.
 			$file_ext  = strtolower( pathinfo( $this->file, PATHINFO_EXTENSION ) );
 			$file_mime = $this->mime_type;
```

Comment:
I think you need to change this now. Based on the comment above, this class property used to always have the MIME type of the input file, and that's what is needed here. But now since it can be overwritten, just accessing `$this->mime_type` here is not correct anymore. I think we now need to also call `$this->get_mime_type( $file_ext )` here, at least in case that `$this->mime_type_set` is true.